builder/builder.go

```diff
@@ -504,14 +504,14 @@ func (b *Builder) createFilesForVolume(ctx context.Context, volMount *volume.Vol
 	return nil
 }
 
-func (b *Builder) populateVolumeWithFiles(ctx context.Context, volMount *volume.VolumeMount) error {
+func (b *Builder) populateVolumeWithFiles(ctx context.Context, volMount *volume.Volume) error {
```

Comment:
Giving this more thought, why do we need to `populateVolumeWithFiles`. Could you remind me?

We are using bind mounts **(not docker volume mount)** anyway. I think we can drop this entire method?

Edit:

When you start creating files for Volume.

createFilesForVolume(), lets create all the files in a folder called `volumeSource`.

So, `mkdir volumeSources` first, and then mkdir your volMount.Name.

This will have a cleaner folder structure. All the volumes will be inside the folder `volumeSources`. Now, 

```
volumes:
  - name: mysecrets
    secret:
      - mysecret1: {{"this is a non encoded string1" | b64enc}}
      - mysecret2: {{"this is a non encoded string2" | b64enc}}
```

After calling `createFilesForVolume`, folder structure on your host machine is:

```
my_acb_home_vol (b.workspaceDir)
|__volumeSources
       |__mysecrets
            |__mysecret1
            |__mysecret2
```

In `context.go` when you get docker run args:

```
if len(volMounts) > 0 {
		for key, val := range volMounts {
			sb.WriteString(" --volume " + "volumeSources/" + key + ":" + val)
		}
	}
```

Just an idea, can this work?

Reply:
talked offline. We need named volumes